test/__snapshots__/test/test.js.md

```diff
@@ -850,6 +889,197 @@ Generated by [AVA](https://ava.li).
     }␊
     `
 
+## patternProperties.1.js
+
+> Snapshot 1
+
+    `/**␊
+     * This file was automatically generated by json-schema-to-typescript.␊
+     * DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,␊
+     * and run json-schema-to-typescript to regenerate this file.␊
+     */␊
+    ␊
+    export interface Parent {␊
+      [k: string]: any;␊
+    }␊
+    /**␊
+     * This interface was referenced by `Parent`'s JSON-Schema definition␊
+     * via the `patternProperty` "^[a-zA-Z]+".␊
+     */␊
+    export interface Child {␊
+      aProperty?: string;␊
+      [k: string]: any;␊
+    }␊
+    `
+
+## patternProperties.2.js
+
+> Snapshot 1
+
+    `/**␊
+     * This file was automatically generated by json-schema-to-typescript.␊
+     * DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,␊
+     * and run json-schema-to-typescript to regenerate this file.␊
+     */␊
+    ␊
+    export interface Parent {␊
+      a?: number;␊
+      b?: string;␊
+      [k: string]: any;␊
+    }␊
+    /**␊
+     * This interface was referenced by `Parent`'s JSON-Schema definition␊
+     * via the `patternProperty` "^[a-zA-Z]+".␊
+     */␊
+    export interface Child {␊
+      aProperty?: string;␊
+      [k: string]: any;␊
+    }␊
+    `
+
+## patternProperties.3.js
+
+> Snapshot 1
+
+    `/**␊
+     * This file was automatically generated by json-schema-to-typescript.␊
+     * DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,␊
+     * and run json-schema-to-typescript to regenerate this file.␊
+     */␊
+    ␊
+    export interface Parent {␊
+      [k: string]: Child;␊
+    }␊
+    /**␊
+     * This interface was referenced by `Parent`'s JSON-Schema definition␊
+     * via the `patternProperty` "^[a-zA-Z]+".␊
+     */␊
+    export interface Child {␊
+      aProperty?: string;␊
+      [k: string]: any;␊
+    }␊
+    `
+
+## patternProperties.4.js
+
+> Snapshot 1
+
+    `/**␊
+     * This file was automatically generated by json-schema-to-typescript.␊
+     * DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,␊
+     * and run json-schema-to-typescript to regenerate this file.␊
+     */␊
+    ␊
+    /**␊
+     * This interface was referenced by `Parent`'s JSON-Schema definition␊
+     * via the `patternProperty` "^[0-9]+".␊
+     */␊
+    export type NumberChild = number;␊
+    /**␊
+     * This interface was referenced by `Parent`'s JSON-Schema definition␊
+     * via the `patternProperty` "^[a-zA-Z]+".␊
+     */␊
+    export type StringChild = string;␊
+    ␊
+    export interface Parent {}␊
```

Comment:
Is this the right output? I would expect

```ts
export interface Parent {
  [k: string]: NumberChild | StringChild
}
```

Reply:
That would be better, but I didn't manage to get that working. For now the set of rules are quite conservative.

Reply:
Tracking this in https://github.com/bcherny/json-schema-to-typescript/issues/160